intro-3d-2023-1/Assets/Scripts/Enemy/FiniteStateMachine/FiniteStateMachine.cs

```diff
@@ -9,7 +9,11 @@ public class FiniteStateMachine : MonoBehaviour
     [Space(10)]
     [SerializeField] private Transform _target;
 
+    [Space(10)]
+    [SerializeField] private GameObject _this;
+
     public Transform Target => _target;
+    public GameObject This => _this;
```

Comment:
Esto no es necesario.
Puedes hacer `fms.transform` o `fms.gameObject` y funciona perfecto.
También, es confuso con el propio `this` de C# y Unity. Para lo que lo estas utilizando, posiblemente no es necesario.

Ahora! Si esa referencia es sobre un objeto diferente (un hijo por ejemplo), entonces usa un mejor nombre que no sea `_this` 😛